common-app-lib/src/cmr/common_app/data/search/collection_for_gran_acls_caches.clj

```diff
@@ -81,15 +81,49 @@
                                                   :collection
                                                   elastic-item))
                           :concept-id (:_id elastic-item)))
-       query (q-mod/query {:concept-type :collection
-                           :condition q-mod/match-all
-                           :skip-acls? true
-                           :page-size :unlimited
-                           :result-format :query-specified
-                           :result-fields (cons :concept-id acl-rhh/collection-elastic-fields)
-                           :result-features {:query-specified {:result-processor result-processor}}})]
+       query (qm/query {:concept-type :collection
+                        :condition qm/match-all
+                        :skip-acls? true
+                        :page-size :unlimited
+                        :result-format :query-specified
+                        :result-fields (cons :concept-id acl-rhh/collection-elastic-fields)
+                        :result-features {:query-specified {:result-processor result-processor}}})]
   (:items (qe/execute-query context query))))
 
+(defn- fetch-collection
+ ([context collection-concept-id]
+  "Fetches one collection from elastic search by concept id. If no collection is found, will return nil."
+ (let [result-processor (fn [_ _ elastic-item]
+                         (assoc (util/delazy-all (acl-rhh/parse-elastic-item
+                                                  :collection
+                                                  elastic-item))
+                          :concept-id (:_id elastic-item)))
+       query (qm/query {:concept-type :collection
+                        :condition (qm/string-condition :concept-id collection-concept-id true false)
+                        :skip-acls? true
+                        :page-size :unlimited
+                        :result-format :query-specified
+                        :result-fields (cons :concept-id acl-rhh/collection-elastic-fields)
+                        :result-features {:query-specified {:result-processor result-processor}}})
+       result (first (:items (qe/execute-query context query)))]
+  result))
+ ([context provider-id entry-title]
```

Comment:
so is the request the same, only difference in how your processing? If so I don't think we should overload the function like this, I think the common part should be made a function that both actions here use. To much redundancy.

Reply:
fixed